zebra-network/src/peer/client.rs

```diff
@@ -28,6 +31,9 @@ pub struct Client {
     ///
     /// `None` unless the connection or client have errored.
     pub(crate) error_slot: ErrorSlot,
+
+    /// The peer's protocol version.
```

Comment:
Nit: the `Client` for a peer connection might not be dropped until 2 minutes after the peer disconnects. That's enough time for a peer to restart, upgrade, and reconnect.

It's also possible to load-balance different versions on the same listener port. Or malicious peers can send whatever versions they want on each connection.

```suggestion
    /// The peer connection's protocol version.
```